content/en/docs/pipelines/installation/localcluster-deployment.md

```diff
@@ -0,0 +1,228 @@
+
+  * [Kind - Known Issues](https://kind.sigs.k8s.io/docs/user/known-issues/)
+
+  * [Kind - Working Offline](https://kind.sigs.k8s.io/docs/user/working-offline/)
+
+### Create your cluster on Kind
+
+Creating a Kubernetes cluster is as simple as ```kind create cluster```.
+
+This will bootstrap a Kubernetes cluster using a pre-built node image - you can find it on docker hub kindest/node. If you desire to build the node image yourself see the building image section. To specify another image use the --image flag.
+
+By default, the cluster will be given the name kind. Use the --name flag to assign the cluster a different context name.
+
+### Set up your cluster on K3S
+
+K3s is a fully compliant Kubernetes distribution with the following enhancements:
+
+* Packaged as a single binary.
+* Lightweight storage backend based on sqlite3 as the default storage mechanism. etcd3, MySQL, Postgres also still available.
+* Wrapped in simple launcher that handles a lot of the complexity of TLS and options.
+* Secure by default with reasonable defaults for lightweight environments.
+* Simple but powerful “batteries-included” features have been added, such as: a local storage provider, a service load balancer, a Helm controller, and the Traefik ingress controller.
+* Operation of all Kubernetes control plane components is encapsulated in a single binary and process. This allows K3s to automate and manage complex cluster operations like distributing certificates.
+* External dependencies have been minimized (just a modern kernel and cgroup mounts needed). K3s packages required dependencies, including:
+     * containerd
+     * Flannel
+     * CoreDNS
+     * CNI
+     * Host utilities (iptables, socat, etc)
+     * Ingress controller (traefik)
+     * Embedded service loadbalancer
+     * Embedded network policy controller
+
+K3s provides an installation script that is a convenient way to install it as a service on systemd or openrc based systems. This script is available at https://get.k3s.io. To install K3s using this method, just run:
+
+```
+curl -sfL https://get.k3s.io | sh -
+```
+Note, kind use containerd as default container-runtime hence you cannot use the standard kubeflow pipeline manifests.
+
+**References**:
+
+  * [K3S - Quick Start Guide](https://rancher.com/docs/k3s/latest/en/quick-start/)
+
+  * [K3S - Known Issues](https://rancher.com/docs/k3s/latest/en/known-issues/)
+
+  * [K3S - FAQ](https://rancher.com/docs/k3s/latest/en/faq/)
+
+### Create your cluster on K3S
+
+Creating a Kubernetes cluster is as simple as ```sudo k3s server &```.
+
+This will bootstrap a Kubernetes cluster kubeconfig is written to /etc/rancher/k3s/k3s.yaml
+
+```
+sudo k3s kubectl get node
+```
+
+On a different node run the below. NODE_TOKEN comes from /var/lib/rancher/k3s/server/node-token on your server
+
+``` 
+sudo k3s agent --server https://myserver:6443 --token ${NODE_TOKEN}
+```
+### Set up your cluster on K3S on WSL2
```

Comment:
_(I'm sharing my refactored section on WSL here, in case you want to use it.)_

_(Feel free to change this.)_

_(Because I can't have "nested" code blocks in Markdown (three back ticks) without breaking the preview here, add them before and after command lines, if necessary.)_

```
### Setting up a cluster on k3s on Windows Subsystem for Linux (WSL)

The Windows Subsystem for Linux (WSL) lets developers run a GNU/Linux environment—
including most command-line tools, utilities, and applications—directly on Windows, unmodified, 
without the overhead of a traditional virtual machine or dualboot setup.

The full instructions for installing WSL can be found on the [official Windows site](https://docs.microsoft.com/en-us/windows/wsl/install-win10).

1. Install WSL

The following steps summarize what you'll need to set up WSL first, and then K3s on WSL.

  1. Install [WSL2] by following the official [docs](https://docs.microsoft.com/en-us/windows/wsl/install-win10).

  2. As per the official instructions, update WSL2 and download your favorite distibution:

     * [SUSE Linux Enterprise Server 15 SP1](https://www.microsoft.com/store/apps/9PN498VPMF3Z)
     * [openSUSE Leap 15.2](https://www.microsoft.com/store/apps/9MZD0N9Z4M4H)
     * [Ubuntu 18.04 LTS](https://www.microsoft.com/store/apps/9N9TNGVNDL3Q)
     * [Debian GNU/Linux](https://www.microsoft.com/store/apps/9MSVKQC78PK6)

**References**:

* [K3s on WSL2: Quick Start Guide](https://gist.github.com/ibuildthecloud/1b7d6940552ada6d37f54c71a89f7d00)
```

Reply:
```
### Creating a cluster on k3s on WSL2

1. To create a Kubernetes cluster, use `sudo ./k3s server`.

This will bootstrap a Kubernetes cluster but you will cannot yet access from your Windows machine to the cluster itself.

**Note:** You can't install k3s using the curl script because there is no supervisor (systemd or openrc) in WSL2.

2. Download k3s binary from https://github.com/rancher/k3s/releases/latest. Then run this command in the directory where you download the k3s binary to:

```
```
chmod +x k3s
```
```

3. Run k3s:

```
```
sudo ./k3s server
```
```

### Setup access to your WSL2 instance

* Copy `/etc/rancher/k3s/k3s.yaml` from WSL to your home in Windows to 
`%HOME%.kube\config`. Edit the copied file and change the server URL from 
`https://localhost:6443` to the IP of the your WSL2 instance (`ip addr show dev eth0`) (For 
example, `https://192.168.170.170:6443`.)

* Run kubectl from windows, if you don't have it yet you may obtain it from 
[here](https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-windows)
```

Reply:
Some steps here should have been sequenced from 1 to 3, for example, so I did that for you. Please keep that in mind.